hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1384,6 +1388,7 @@ private void setMembersRef(Map<Address, MemberImpl> memberMap) {
 
     @Override
     public void shutdown(boolean terminate) {
+        membershipEventExecutor.shutdown();
```

Comment:
with above mentioned solution, you don't need to handle the life-cycle of the executor